src/ngx_http_modsecurity_module.c

```diff
@@ -622,6 +625,20 @@ ngx_http_modsecurity_create_main_conf(ngx_conf_t *cf)
     msc_set_connector_info(conf->modsec, MODSECURITY_NGINX_WHOAMI);
     msc_set_log_cb(conf->modsec, ngx_http_modsecurity_log);
 
+    /* Set up audit log reload */
+    ngx_str_t log_reload_file = ngx_string("/dev/null");
+    conf->audit_log_reload = ngx_conf_open_file(cf->cycle, &log_reload_file);
+    if (conf->audit_log_reload == NULL) {
+        dd("failed to open file for triggering log reload");
+        return NGX_CONF_ERROR;
+    }
+    conf->audit_log_reload->data = ngx_list_create(cf->pool, 100, sizeof(Rules*));
+    if (conf->audit_log_reload->data == NULL) {
+        dd("failed to create list of rule sets for log reload");
+        return NGX_CONF_ERROR;
+    }
+    conf->audit_log_reload->flush = ngx_http_modsecurity_log_reload;
```

Comment:
This is the mechanism by which we trigger reopen in response to `nginx -s reopen` and SIGUSR1.